lib/ts/recipe/emailverification/recipe.ts

```diff
@@ -164,4 +186,28 @@ export default class Recipe extends RecipeModule {
     isErrorFromThisRecipe = (err: any): err is STError => {
         return STError.isErrorFromSuperTokens(err) && err.fromRecipe === Recipe.RECIPE_ID;
     };
+
+    getEmailForUserId = async (userId: string, userContext: any): Promise<string> => {
+        if (this.config.getEmailForUserId !== undefined) {
+            const email = await this.config.getEmailForUserId(userId, userContext);
+            if (email !== undefined) {
+                return email;
+            }
+        }
+
+        for (const getEmailForUserId of this.getEmailForUserIdFuncsFromOtherRecipes) {
+            try {
+                const email = await getEmailForUserId(userId, userContext);
+                return email;
+            } catch {
+                // We ignore these, they should all be: Unknown User ID provided
```

Comment:
Not unless there is a connection issue with the db or something. So you might want to change the return type of `getEmailForUserId` to be something like `Promise<{status: "OK",  email: string} | {status: "UNKNOWN_USER_ID_ERROR" | "NO_EMAIL_EXISTS_FOR_USER_ERROR"}>`.

The `NO_EMAIL_EXISTS_FOR_USER_ERROR` would be useful for passwordless recipe when the userId exists, but they have a phone number only associated with them.

Reply:
We should discuss this tomorrow.